### PR TITLE
fix(test): smoke testing env and config overrides

### PR DESCRIPTION
**Pull request by razvan**
# Description

Part of: https://github.com/stackabletech/issues/issues/548

:green_circle: Test

```shell
$ ./scripts/run-tests --test-suite smoke-latest --skip-release
...
--- PASS: kuttl (209.68s)
    --- PASS: kuttl/harness (0.00s)
        --- PASS: kuttl/harness/smoke_postgres-12.5.6_hive-3.1.3_openshift-false_s3-use-tls-false (208.28s)
PASS
```

## Definition of Done Checklist

- Not all of these items are applicable to all PRs, the author should update this template to only leave the boxes in that are relevant
- Please make sure all these things are done and tick the boxes

```[tasklist]
# Author
- [ ] Changes are OpenShift compatible
- [ ] CRD changes approved
- [ ] CRD documentation for all fields, following the [style guide](https://docs.stackable.tech/home/nightly/contributor/docs/style-guide).
- [ ] Helm chart can be installed and deployed operator works
- [ ] Integration tests passed (for non trivial changes)
- [ ] Changes need to be "offline" compatible
```

```[tasklist]
# Reviewer
- [ ] Code contains useful comments
- [ ] Code contains useful logging statements
- [ ] (Integration-)Test cases added
- [ ] Documentation added or updated. Follows the [style guide](https://docs.stackable.tech/home/nightly/contributor/docs/style-guide).
- [ ] Changelog updated
- [ ] Cargo.toml only contains references to git tags (not specific commits or branches)
```

```[tasklist]
# Acceptance
- [ ] Feature Tracker has been updated
- [ ] Proper release label has been added
- [ ] [Roadmap](https://github.com/orgs/stackabletech/projects/25/views/1) has been updated
```

### CODE REVIEW

File: tests/templates/kuttl/smoke/60-install-hive.yaml.j2 [hunk 1]
```diff
@@ -26,9 +26,24 @@ spec:
     config:
       logging:
         enableVectorAgent: {{ lookup('env', 'VECTOR_AGGREGATOR') | length > 0 }}
+    envOverrides:
+      COMMON_VAR: role-value # overridden by role group below
+      ROLE_VAR: role-value   # only defined here at role level
+    configOverrides:
```
Comment (xeniape) [inline tests/templates/kuttl/smoke/60-install-hive.yaml.j2:32]: I would suggest adding some test cases covering the whole override cycle. For example adding
```
configOverrides:
  hive-site.xml:
    hive.metastore.metrics.enabled: "false"
```
to the HiveCluster and the test
```
- script: |
      kubectl -n $NAMESPACE get cm hive-metastore-default -o yaml | yq -e '.data."hive-site.xml"' | grep "<name>hive.metastore.metrics.enabled</name>" -A 1 | grep "<value>false</value>"
```
to the assert, would reproduce the error of configOverrides not working properly.

**Review by xeniape — CHANGES_REQUESTED** [review]
Some small suggestions:
- Bumping stackable-operator version to 0.74.0 (containing the fix for the override issue mentioned in the other comment)
- Adding an immutable recapture to `pod_template` to have a bit of a safety net regarding changes done to the template after overrides were applied, see https://github.com/stackabletech/airflow-operator/pull/490/files#diff-0561a03543e878867e5068ca78d8064c90cb4ad0b7f8d3e36a157e9820096eb2R972

File: tests/templates/kuttl/smoke/60-install-hive.yaml.j2 (hunk 1 above)
Comment (razvan) [reply tests/templates/kuttl/smoke/60-install-hive.yaml.j2:32]: I don't understand. This is exactly what is tested with `hive-env.sh`.

Comment (razvan) [reply tests/templates/kuttl/smoke/60-install-hive.yaml.j2:32]: I'm a bit meh on the shadowing thing. It feels very naive. One can go around it any time by using a different var.
I would prefer to make `merge_from` composable.

Updated the op-rs version here: https://github.com/stackabletech/hive-operator/pull/507/commits/62e0b97b3f2fd48aa229bef456e1ce97e401e5a8

Comment (xeniape) [reply tests/templates/kuttl/smoke/60-install-hive.yaml.j2:32]: > I'm a bit meh on the shadowing thing. It feels very naive. One can go around it any time by using a different var. I would prefer to make `merge_from` composable.

Just a note: For now we decided to leave that out and address the merging composability during later refactorings.

Comment (razvan) [reply tests/templates/kuttl/smoke/60-install-hive.yaml.j2:32]: Updated test: https://github.com/stackabletech/hive-operator/pull/507/commits/cf3a537f1918626058557aba70ef2d649cf85432

:green_circle: Test

```
--- PASS: kuttl (136.86s)
    --- PASS: kuttl/harness (0.00s)
        --- PASS: kuttl/harness/smoke_postgres-12.5.6_hive-3.1.3_openshift-false_s3-use-tls-false (136.20s)
PASS
```

**Review by xeniape — APPROVED** [review]

**Pull request merged**